### PR TITLE
[17197] Transport descriptor API

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
This PR:

* Updates project name
* Removes Fast DDS as dependency
* Adds Transport Descriptor module API

Caveats:

1. Some of the documented `element_invalid` exceptions should be verified against the XSD schema to ensure that only methods which values are validated throw this exception.
1. Before merging this PR it would be nice to include the code style job in the Github actions.

### CODE REVIEW

**Comment by JLBuenoLopez-eProsima** [comment]
Closed and reopened so the new linter workflow runs

File: lib/include/TransportDescriptor.hpp [hunk 1]
```diff
@@ -0,0 +1,2033 @@
...
+/************************************************************************/
+
+/**
+ * @brief Set the Transport Descriptor kind.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ * @param[in] kind Transport descriptor kind.
+ *
+ * @throw element_invalid Exception if the provided kind is not valid.
```
Comment (JesusPoderoso) [inline lib/include/TransportDescriptor.hpp:1438]: I think that all `set`functions should also throw the `file_not_found` and `element_not_found` exceptions.

File: lib/include/TransportDescriptor.hpp [hunk 2]
```diff
@@ -0,0 +1,2033 @@
...
+
+/**
+ * @brief Append an IP address to the whitelisted network interfaces collection.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ * @param[in] ip_address IP address to be included in the whitelist collection.
+ *
+ * @throw element_invalid Exception if the provided IP address is not valid.
+ */
```
Comment (JesusPoderoso) [inline lib/include/TransportDescriptor.hpp:1882]: As in `set` functions, I think that all `push` and `update` functions should also throw the `file_not_found` and `element_not_found` exceptions.

File: lib/include/TransportDescriptor.hpp [hunk 3]
```diff
@@ -0,0 +1,2033 @@
...
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ *
+ * @return std::string Specific TCP Transport Descriptor TLS password.
+ *
+ * @throw file_not_found Exception if the provided XML file is not found/readable.
+ * @throw element_not_found Exception if the specified transport descriptor profile is not found in the XML file or the
+ *        TLS password element has not been set in the profile.
+ */
+std::string print_tls_password(
```
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply lib/include/TransportDescriptor.hpp:381]: In this case that it is only one complex element within the TransportDescriptor, and the complex element name is not that lengthy, we agreed to keep the name in the method instead of creating a new namespace. @EduPonz, what do you think?

File: lib/include/TransportDescriptor.hpp (hunk 1 above)
Comment (JLBuenoLopez-eProsima) [reply lib/include/TransportDescriptor.hpp:1438]: This is not correct, if the file is not found the library should create a new XML file. And if the element is not found, the library should create it with the `set` command. The only case where the `element_not_found` exception should be thrown is in case of updating a list/map element.

File: lib/include/TransportDescriptor.hpp (hunk 2 above)
Comment (JLBuenoLopez-eProsima) [reply lib/include/TransportDescriptor.hpp:1882]: Again, the `push` element should include the element or overwrite it if found. But it should not throw the exception. The `update` throws `element_not_found` because it is required when updating that the element can be found.

**Comment by JLBuenoLopez-eProsima** [comment]
@JesusPoderoso, I was specially interested on knowing which `set` commands can throw `element_invalid` taking into account what can be validated with the XSD Schema. With that information we can really fix which `set` functions could throw that Exception and I can move forward with the tests.

**Comment by JesusPoderoso** [comment]
@JLBuenoLopez-eProsima Technically, the XML libraries would receive as an input for any element value a `string`, and the schema is in charge of trying to parse that input into the expected type of data, and report the possible parsing type error.
In that way, and simplifying all methods, we may change ALL inputs into `string`, and do not be worried about parsing any input between the source (CLI / GUI / whatever) and the library API.

Anyway, besides number parsing (`int` / `uint` / `float` / etc, which is a straightforward parse error), all `string` must not be empty, and the `string` with specific format would have a regexp to chek it.
In conclusion, if all imputs are `string`, all set functions can throw the `element_invalid` exception. Otherwise if all inputs already have the expected datatype, only some of the `string` inputs would throw that exception (and some range-limited number would too, as 0 <= domainId <= 232 ). Shall I provide you a list with all of them?

**Comment by JLBuenoLopez-eProsima** [comment]
@EduPonz, what do you think? Do we change every `set` type to `string`? I think that leaving the responsibility to the XML library is cool.

File: lib/include/TransportDescriptor.hpp [hunk 4]
```diff
@@ -0,0 +1,2033 @@
...
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ *
+ * @return uint32_t Transport descriptor send buffer size.
+ *
+ * @throw file_not_found Exception if the provided XML file is not found/readable.
+ * @throw element_not_found Exception if the specified transport descriptor profile is not found in the XML file or the
+ *        send buffer size element has not been set in the profile.
+ */
```
[the remark this answers is not in the document]
Comment (EduPonz) [reply lib/include/TransportDescriptor.hpp:86]: I would have more than one exception, but I'd also keep the what as a help of sorts.

File: lib/include/TransportDescriptor.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (EduPonz) [reply lib/include/TransportDescriptor.hpp:381]: That is what we agreed, yes

File: lib/include/TransportDescriptor.hpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (jsantiago-eProsima) [reply lib/include/TransportDescriptor.hpp:86]: I do not mind having more than one exception either. I think this will also be at least a partially reliant on what functionalities the underlying libraries provide for error handling.

**Comment by EduPonz** [comment]
> @EduPonz, what do you think? Do we change every `set` type to `string`? I think that leaving the responsibility to the XML library is cool.

I definitely agre with forwarding the responsibility as much as possible

File: lib/include/TransportDescriptor.hpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (JesusPoderoso) [reply lib/include/TransportDescriptor.hpp:86]: This decition would be taken based on the exceptions thrown by the final XML library used.

Last tests would be performed to determine the library.

[the remark this answers is not in the document]
Comment (JesusPoderoso) [reply lib/include/TransportDescriptor.hpp:88]: Finally it was determined that both libraries do not throw multiple exceptions neither the complete set of exceptions designed, so we should implement a Exception class to throw them.

**Comment by JesusPoderoso** [comment]
LGTM
